Tests/test_file_ppm.py

```diff
@@ -50,6 +50,55 @@ def test_pnm(tmp_path):
             assert_image_equal(im, reloaded)
 
 
+def test_not_ppm(tmp_path):
+    path = str(tmp_path / "temp.djvurle")
+    with open(path, "wb") as f:
+        f.write(b"PyInvalid")
+
+    with pytest.raises(UnidentifiedImageError):
+        with Image.open(path):
+            pass
```

Comment:
So this is testing that not any file starting with P is considered a PPM file?

Reply:
Almost. Not any file starting with `Py` (accounting for some stated "PIL extensions").

In my other PR for the plain PPM decoder (unhappily built on top of this one), I replaced that test with:
```python
def test_not_ppm(tmp_path):
    with pytest.raises(SyntaxError):
        PpmImagePlugin.PpmImageFile(fp=BytesIO(b"PyInvalid"))